### ISSUE TITLE
[Suggestion] The Forcefields Awakens

### ISSUE THREAD

**Issue by ZachZent** [tsunamayo/Starship-EVO#4878]
This has popped up every once in awhile over the years, but is there any plans for forcefields? Projected beam wall with a selector to allow or prevent people/builds through. Maybe the "walk though mode" can be a more clear texture while the "blocking mode" can have hexes. The forcefields can also be turned on or off of course. For balance sake since this came up in the Discord, to prevent a case of "shield stacking", forcefields should require power to function, but not recharge like shields.

![image](https://user-images.githubusercontent.com/36980446/185732291-20f84d52-2137-4ed8-8213-c6eddcdcd8d3.png)
Made with spotlights which looks really cool

1. Simple option: An X & Y axis (no Z) stretched block, similar to fences, that will create a single forcefield (with blocks on the edges so it isn't just a floating glowey wall). Benefits it that it would be easy to place and wouldn't require an entire new set of configuration like CEs. Downside is that it will always be stuck to the existing brick shapes. Still, this would be (in theory) the fastest solution and satisfy most requirements
![image](https://user-images.githubusercontent.com/36980446/185731896-7e65376a-3dc2-4279-9ac2-5b58b7ff6ec2.png)

2. A single "CE" that will project the wall as far as the set configuration allows. Single stretched block, only one axis instead of two, and set the wall to project however far. Just like the spotlight. This could be an issue with forcefields clipping into walls, but it doesn't have the restrictions of shapes
![image](https://user-images.githubusercontent.com/36980446/185732676-c6004ddb-89bc-4f52-a518-df13d6c5bb46.png)
(There are not many examples lol)

3. One item you have to place down twice. If they are opposite of each other, then a wall is projected. This would be the most complicated, but it would allow unique shapes.
![image](https://user-images.githubusercontent.com/36980446/185729976-bc24cce8-6689-42dc-9b93-e224adc90a47.png)
Some nostalgia from Xerma

**Comment by tsunamayo** [comment]
@ZachZent Yes it is still in my to-do list. There are a few option indeed, could be also one side where you set the height manually.
Do you want a forcefield to hold air only (so purely cosmetic atm), or something that also block passage and ship to get through?

Thanks

**Comment by ZachZent** [comment]
@tsunamayo Maybe as a placeholder, fully cosmetic. Full docking and air isn't in game yet so it isn't a priority of course.

As for something to block passage and ships, a simple switch to turn entry on and off. Maybe a slight cosmetic change to signify the change.

Ships can pass through: Watery
![image](https://user-images.githubusercontent.com/36980446/199903039-9839f2e3-1ec0-436d-910c-2205c5c24587.png)


Ship's can't pass through: Solid light
![image](https://user-images.githubusercontent.com/36980446/199902639-ef2392bb-1c9c-46c6-92eb-079b8d07114e.png)

**Comment by Kaiser-Indrasil** [comment]
One more inspiration from yours truly.

![Stevo_Astero_Drone_Bay_Force_Field-1.gif](https://user-images.githubusercontent.com/63756057/199934544-d2a9a78f-6835-4051-a1fb-ea402b18d900.gif)

**Comment by tsunamayo** [comment]
Thanks a lot for those ideas. Great use of the flood light @Kaiser-Indrasil !
